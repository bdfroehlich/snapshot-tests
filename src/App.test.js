import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer'


//Jest will look in a special directory, __snapshots__, for a snapshot that matches the test file currently running
test('renders a snapshot', () => {
  //pass a React element in JSX form, App, to renderer.create. 
  //renderer.create renders the element and creates a special output that has a toJSON method
  const tree = renderer.create(<App />).toJSON()
  // console.log(tree)
  //compares the new snapshot in tree to the saved snapshot if one already exists
  expect(tree).toMatchSnapshot()
})
