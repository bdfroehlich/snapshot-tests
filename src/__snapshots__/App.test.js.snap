// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <div
    className="card bg-secondary"
    style={
      Object {
        "width": "25rem",
      }
    }
  >
    <img
      className="card-img-top"
      src="https://avatars.githubusercontent.com/u/89557567?v=4"
      style={
        Object {
          "width": "25rem",
        }
      }
    />
    <div
      className="card-body"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className="card-title h5"
        style={
          Object {
            "color": "blanchedalmond",
            "fontFamily": "arial",
            "fontSize": "30px",
            "textAlign": "center",
          }
        }
      >
        Benjamin Froehlich
      </div>
      <p
        className="card-text"
        style={
          Object {
            "color": "blanchedalmond",
            "fontFamily": "arial",
            "fontSize": "15px",
            "fontWeight": "normal",
            "textAlign": "center",
          }
        }
      >
        I am currently a student in the KSU Full Stack Web Development Program and I am excited to get to the deployment phase of my new career.
      </p>
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <a
          className="btn btn-info"
          disabled={false}
          href="https://github.com/bdfroehlich"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Visit my GitHub
        </a>
      </div>
    </div>
  </div>
</div>
`;
